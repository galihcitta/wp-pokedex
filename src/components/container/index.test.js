import React from "react";
import Container from "./index";

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Container data-test="container-test-mount" />);
});

describe("Container Component", () => {
    it("renders and create snapshot <Content />", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders", () => {
        expect(wrapper.exists()).toBe(true);
    });

    it("check no padding", () => {
        expect(
            renderer
                .create(
                    <Container noPadding data-test="container-test-shallow" />,
                )
                .toJSON(),
        ).toHaveStyleRule("padding", "0");
    });

    it("check with padding", () => {
        expect(
            renderer
                .create(<Container data-test="container-test-shallow" />)
                .toJSON(),
        ).toHaveStyleRule("padding", "20px 15px 10px 15px");
    });
});
