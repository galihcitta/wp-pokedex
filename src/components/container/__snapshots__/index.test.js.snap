// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders and create snapshot <Content /> 1`] = `
.c0 {
  background-color: #f7f7f7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 auto;
  padding: 20px 15px 10px 15px;
  padding-bottom: 110px;
  width: 450px;
}

@media (max-width:450px) {
  .c0 {
    width: 100%;
  }
}

<div
  className="c0"
  data-test="container-test-mount"
/>
`;
